fix: show only goal if balance is 0

sajald77 opened the pull request:
resolves GYS-7802

Agill-Sheron commented:
I think this might be a duplicate as I worked for this in Goal fixes: https://linear.app/geyser/issue/GYS-7806/goal-fixes

sajald77 commented:
> I think this might be a duplicate as I worked for this in Goal fixes: https://linear.app/geyser/issue/GYS-7806/goal-fixes

Yes, turns out it is, fixed. but still I gave it a little bit of a refactor to make things clearer, when an outside eyes watch it. 

I added a couple more variables and made it simpler to understand, I hope. :D

Agill-Sheron commented:
> > I think this might be a duplicate as I worked for this in Goal fixes: https://linear.app/geyser/issue/GYS-7806/goal-fixes
> 
> Yes, turns out it is, fixed. but still I gave it a little bit of a refactor to make things clearer, when an outside eyes watch it.
> 
> I added a couple more variables and made it simpler to understand, I hope. :D

I agree, it's way more clear like this, thanks man!
